.gitlab/container_build/docker_linux.yml

```diff
@@ -119,6 +119,49 @@ docker_build_agent7:
+#   artifacts:
+#     expire_in: 1 week
+#     paths:
+#       - "${CI_COMMIT_SHA}-baseline_sha"
+#     when: always
+#   script:
+#     - git fetch origin
+#     - echo -n $(git merge-base ${CI_COMMIT_SHA} origin/main) > "${CI_COMMIT_SHA}-baseline_sha"
+#     - SMP_ACCOUNT_ID=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.single-machine-performance-account-id --with-decryption --query "Parameter.Value" --out text)
+#     - SMP_AGENT_TEAM_ID=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.single-machine-performance-agent-team-id --with-decryption --query "Parameter.Value" --out text)
+#     - aws configure set aws_access_key_id $(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.single-machine-performance-bot-access-key-id --with-decryption --query "Parameter.Value" --out text) --profile single-machine-performance
+#     - aws configure set aws_secret_access_key $(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.single-machine-performance-bot-access-key --with-decryption --query "Parameter.Value" --out text) --profile single-machine-performance
+#     - aws configure set region us-west-2 --profile single-machine-performance
+#     - aws s3 cp --profile single-machine-performance --only-show-errors "${CI_COMMIT_SHA}-baseline_sha" "s3://${SMP_AGENT_TEAM_ID}-smp-artifacts/information/"
+#     # NOTE(geoffrey.oxberry@datadoghq.com): For the purposes of standing up a
+#     # minimum viable product, assume a container has already been built for the
+#     # merge base -- this assumption is actually true if the target branch of the
+#     # pull request is `main`. If not, we need to (sort of) extend
+#     # .docker_build_on_job_definition, .docker_build_job_definition_amd64, and
+#     # docker_build_agent7. Using the `extends:` keyword won't actually work
+#     # because it'll reverse deep merge settings, so instead this hypothetical
+#     # build task could use `!reference` tags liberally, except for the `script`
+#     # keyword, which would be some suitably modified version of the contents of
+#     # `.docker_build_on_job_definition:script`. As far as I can tell, we only need
+#     # to build a container if the target branch is *not* `main`; it may make sense
+#     # to simply exclude that case.
+#     #
+#     # If we need to build that container, we probably also need to publish it,
+#     # which could be done by suitably modifying a *renamed copy* of
+#     # `single_machine_performance-amd64-a7` (because we still need the existing
+#     # version of that task). If we need to inject a different version of the
+#     # shell variables that relies in some way on outputs from another SMP job in
+#     # Agent CI, we can use the `before_script` hook to `export` the variable
+#     # with the value we need.
```

Comment:
If we were to restrict the base branch of a pull request to `main`, would that suffice? If so, I think this pull request, as of https://github.com/DataDog/datadog-agent/pull/17449/commits/8dd9f11a9ad8417b3b5622671f95f232f4f7d562, is pretty close to the desired end state of this PR. If not, I'll have some additional questions on building and publishing containers.

Reply:
I think we can restrict to `main` and "stable branches", which are `7.45.x`, `7.44.x` etc.

Reply:
Makes sense to me. @scottopell has pointed out some [recent regression detector results](https://github.com/datadog/datadog-agent/pull/17160#issuecomment-1549872166) that would benefit from setting the baseline to a merge base with `main`, so I'd like to limit this PR to that case, then augment that case with an implementation for other stable branches in a subsequent PR. For other stable branches, we'd need to figure out whether a merge base with a branch like `7.45.x` would require building and publishing a container for that merge base. I'm hoping the other operations needed are straightforward (querying the GitHub API for the PR's base branch, checking that this base branch is a stable branch).